### PR TITLE
Added tests for background tasks

### PR DESCRIPTION
**Pull request by engelke**

### CODE REVIEW

File: background/app/main_test.py [hunk 1]
```diff
@@ -11,3 +11,83 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+import json
+import main
+import pytest
+
+
+class MockFirestoreClient():
```
Comment (tbpg) [inline background/app/main_test.py]: Guidance I've seen for samples is to use the real service for testing.

Comment (engelke) [reply background/app/main_test.py]: I'll look into this and see what I can do. I hadn't seen that, and figured that the point here wasn't to test the APIs or libraries, just the sample code. I'll have to set up a project or find out which one is already being used.

Comment (engelke) [reply background/app/main_test.py]: Tests use the real service for testing now.

File: background/function/main_test.py [hunk 2]
```diff
@@ -11,3 +11,57 @@
...
 # limitations under the License.
+
+import base64
+import json
+import os
+import pytest
+
+from google.cloud import firestore
+
+os.environ['GOOGLE_CLOUD_PROJECT'] = FIRESTORE_PROJECT_ID
```
Comment (tbpg) [inline background/function/main_test.py]: Where does this variable come from? Will the credentials path below work for anyone? Can it come from an environment variable (set in https://github.com/GoogleCloudPlatform/getting-started-python/blob/master/.kokoro/system_tests.sh)?

Comment (engelke) [reply background/function/main_test.py]: No, it doesn't seem to pick up on variables set in system_tests.sh, they need to be set in Kokoro configs. It is now being set there.

File: background/app/main_test.py [hunk 3]
```diff
@@ -11,3 +11,100 @@
...
+from google.cloud import firestore
+from google.cloud import pubsub
+
+os.environ['GOOGLE_CLOUD_PROJECT'] = os.environ['FIRESTORE_PROJECT_ID']
+os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = os.path.realpath(
+    os.path.join(
+        os.path.dirname(__file__),
+        '..',
+        '..',
+        'firestore-service-account.json'
```
Comment (tbpg) [inline background/app/main_test.py]: Can we add the service account from the standard `service-account.json` to the `FIRESTORE_PROJECT_ID` IAM? Then we wouldn't need a separate Firestore service account at all.

If not, can we create a separate `FIRESTORE_GOOGLE_APPLICATION_CREDENTIALS` environment variable?

I'm concerned about hardcoding `../../firestore-service-account.json`.

Comment (tbpg) [reply background/app/main_test.py]: After offline discuss, this sounds good. Can clean up/update at a later date.

**Pull request merged**